### PR TITLE
Revert "Remove IceGrid SessionTimeout properties (#2076)" 

### PR DESCRIPTION
**Pull request by bernardnormier**
This PR reverts an earlier IceGrid PR.

### CODE REVIEW

File: java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/SessionKeeper.java [hunk 1]
```diff
@@ -371,6 +376,8 @@ public void registerObservers() throws Throwable {
         private final AdminSessionPrx _session;
         private final boolean _routed;
 
+        private java.util.concurrent.Future<?> _keepAliveFuture;
```
Comment (bernardnormier) [inline java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/SessionKeeper.java:379]: Something is still missing after this revert - keepAliveFuture isn't assigned anywhere.

File: cpp/src/IceGrid/InternalRegistryI.cpp [hunk 2]
```diff
@@ -35,9 +35,8 @@ InternalRegistryI::InternalRegistryI(
       _session(session)
 {
     auto properties = database->getCommunicator()->getProperties();
-    // TODO: temporary. For now, synchronized with the default idle timeout.
```
Comment (bernardnormier) [inline cpp/src/IceGrid/InternalRegistryI.cpp:38]: Need to revert more updates to IceGrid.

File: java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/SessionKeeper.java (hunk 1 above)
Comment (bernardnormier) [reply java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/SessionKeeper.java:379]: I think this was leftover dead code. I will remove it in a follow-up PR.

File: cpp/src/IceGrid/SessionI.cpp [hunk 3]
```diff
@@ -63,6 +63,24 @@ BaseSessionI::BaseSessionI(const string& id, const string& prefix, const shared_
     }
 }
 
+void
+BaseSessionI::keepAlive(const Ice::Current& current)
+{
+    lock_guard lock(_mutex);
+    if (_destroyed)
+    {
+        throw Ice::ObjectNotExistException(__FILE__, __LINE__, current.id, "", "");
```
Comment (pepone) [inline cpp/src/IceGrid/SessionI.cpp:72]: Is this the correct way to throw ONE? I thought we didn't need to set the id.

File: cpp/src/IceGrid/SessionI.cpp [hunk 4]
```diff
@@ -63,6 +63,24 @@ BaseSessionI::BaseSessionI(const string& id, const string& prefix, const shared_
...
+    }
+
+    _timestamp = chrono::steady_clock::now();
+
+    if (_traceLevels->session > 1)
+    {
+        Ice::Trace out(_traceLevels->logger, _traceLevels->sessionCat);
+        out << _prefix << " session `" << _id << "' keep alive";
```
Comment (pepone) [inline cpp/src/IceGrid/SessionI.cpp:80]:
```suggestion
        out << _prefix << " session '" << _id << "' keep alive";
```

File: java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/SessionKeeper.java (hunk 1 above)
Comment (pepone) [reply java/src/IceGridGUI/src/main/java/com/zeroc/IceGridGUI/SessionKeeper.java:379]: In the PR you reverted, there is a comment that this was a leftover from a previous PR. It is never set in the 3.7 branch.

File: slice/IceGrid/Admin.ice [hunk 5]
```diff
@@ -895,4 +896,5 @@ module IceGrid
         FileIterator * openRegistryStdOut(string name, int count)
             throws FileNotAvailableException, RegistryNotExistException, RegistryUnreachableException;
     }
+
```
Comment (pepone) [inline slice/IceGrid/Admin.ice:899]: extra new line?

**Review by pepone — APPROVED** [review]

File: cpp/src/IceGrid/SessionI.cpp (hunk 3 above)
Comment (bernardnormier) [reply cpp/src/IceGrid/SessionI.cpp:72]: That's indeed incorrect.

File: config/PropertyNames.xml [hunk 6]
```diff
@@ -278,11 +278,14 @@
...
         <property name="Registry.Internal" class="ObjectAdapter" languages="cpp" />
         <property name="Registry.LMDB" class="LMDB" languages="cpp"/>
+        <property name="Registry.NodeSessionTimeout" languages="cpp" default="30" />
         <property name="Registry.PermissionsVerifier" class="Proxy" languages="cpp" />
         <property name="Registry.ReplicaName" languages="cpp" default="Master" />
+        <property name="Registry.ReplicaSessionTimeout" languages="cpp" default="30" />
         <property name="Registry.Server" class="ObjectAdapter" languages="cpp" />
         <property name="Registry.SessionFilters" languages="cpp" default="0" />
         <property name="Registry.SessionManager" class="ObjectAdapter" languages="cpp" />
+        <property name="Registry.SessionTimeout" languages="cpp" default="30" />
```
Comment (externl) [inline config/PropertyNames.xml:288]: The previous settings were 60. I assume we want 30 now?

**Review by externl — APPROVED** [review]

File: config/PropertyNames.xml (hunk 6 above)
Comment (bernardnormier) [reply config/PropertyNames.xml:288]: 30 is what we document.

I will remove Registry.SessionTimeout in a follow-up PR so it's a moot point.

File: slice/IceGrid/Admin.ice (hunk 5 above)
Comment (bernardnormier) [reply slice/IceGrid/Admin.ice]: Removed

File: cpp/src/IceGrid/SessionI.cpp (hunk 3 above)
Comment (bernardnormier) [reply cpp/src/IceGrid/SessionI.cpp]: Fixed

**Pull request merged**